examples/src/main/java/com/example/MazeFuzzer.java

```diff
@@ -0,0 +1,147 @@
+import com.code_intelligence.jazzer.api.Jazzer;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+// A fuzz target that shows how manually informing the fuzzer about important state can make a fuzz
+// target much more effective.
+// This is a Java version of the famous "maze game" discussed in
+// "IJON: Exploring Deep State Spaces via Fuzzing", available at:
+// https://wcventure.github.io/FuzzingPaper/Paper/SP20_IJON.pdf
+public final class MazeFuzzer {
+  private static final String[] MAZE_STRING = new String[] {
+      "  ███████████████████",
+      "    █ █ █   █ █     █",
+      "█ █ █ █ ███ █ █ █ ███",
+      "█ █ █   █       █   █",
+      "█ █████ ███ ███ █ ███",
+      "█       █   █ █ █   █",
+      "█ ███ ███████ █ ███ █",
+      "█ █     █ █     █   █",
+      "███████ █ █ █████ ███",
+      "█   █       █     █ █",
+      "█ ███████ █ ███ ███ █",
+      "█   █     █ █ █   █ █",
+      "███ ███ █ ███ █ ███ █",
+      "█     █ █ █   █     █",
+      "█ ███████ █ █ █ █ █ █",
+      "█ █         █ █ █ █ █",
+      "█ █ █████████ ███ ███",
+      "█   █   █   █ █ █   █",
+      "█ █ █ ███ █████ ███ █",
+      "█ █         █        ",
+      "███████████████████ #",
+  };
+
+  private static final char[][] MAZE = parseMaze();
+  private static final char[][] REACHED_FIELDS = parseMaze();
+
+  public static void fuzzerTestOneInput(byte[] commands) {
+    executeCommands(commands, (x, y, won) -> {
+      if (won) {
+        throw new TreasureFoundException(commands);
+      }
+      // This is the key line that makes this fuzz target work: It instructs the fuzzer to track
+      // every new combination of x and y as a new feature. Without it, the fuzzer would be
+      // completely lost in the maze as guessing an escaping path by chance is close to impossible.
+      Jazzer.exploreState(hash(x, y), 0);
+      if (REACHED_FIELDS[x][y] == ' ') {
+        // Fuzzer reached a new field in the maze, print its progress.
+        REACHED_FIELDS[x][y] = '.';
+        System.out.println(renderMaze(REACHED_FIELDS));
+      }
+    });
+  }
+
+  // Hash function with good mixing properties published by Thomas Mueller
+  // under the terms of CC BY-SA 4.0 at
+  // https://stackoverflow.com/a/12996028
+  // https://creativecommons.org/licenses/by-sa/4.0/
+  private static byte hash(byte x, byte y) {
+    int h = (x << 8) | y;
+    h = ((h >> 16) ^ h) * 0x45d9f3b;
+    h = ((h >> 16) ^ h) * 0x45d9f3b;
+    h = (h >> 16) ^ h;
+    return (byte) h;
+  }
+
+  private static class TreasureFoundException extends RuntimeException {
+    TreasureFoundException(byte[] commands) {
+      super(renderPath(commands));
+    }
+  }
+
+  private static void executeCommands(byte[] commands, Consumer3<Byte, Byte, Boolean> callback) {
+    byte x = 0;
+    byte y = 0;
+    callback.accept(x, y, false);
+
+    for (byte command : commands) {
+      byte nextX = x;
+      byte nextY = y;
+      switch (command % 4) {
```

Comment:
I just added one commit to make it a bit more challenging for the fuzzer. WDYT?

Reply:
Looks good, made one change.